src/NodeGroup/index.d.ts

```diff
@@ -1,20 +1,24 @@
 import * as React from "react";
 import {
-  HashMap,
+  Config,
 } from 'kapellmeister';
 import { GetInterpolator } from '..'
 
-export interface INodeGroupProps {
-  data: Array<any>;
-  keyAccessor: (data: any, index: number) => string | number;
+export type AddArrayLike<T> = unknown extends T ? Config | Config[] : {
+  [P in keyof T]: T[P] | T[P][]
+}
+
+export interface INodeGroupProps<T = unknown, State = unknown> {
+  data: T[];
+  keyAccessor: (data: T, index: number) => string | number;
   interpolation?: GetInterpolator;
-  start: (data: any, index: number) => HashMap;
-  enter?: (data: any, index: number) => (HashMap | Array<HashMap>);
-  update?: (data: any, index: number) => (HashMap | Array<HashMap>);
-  leave?: (data: any, index: number) => (HashMap | Array<HashMap>);
-  children: (nodes: Array<any>) => React.ReactElement<any>;
+  start: (data: T, index: number) =>  unknown extends State ? Config : State;
```

Comment:
I think the return from start needs to be the user supplied or HashMap from Kapellmeister.  The state is not a Config object with timing etc.

```js
  start: (data: T, index: number) =>  unknown extends State ? Config : State;
```

Should be something like:
```js
import {
  HashMap,
  Config,
} from 'kapellmeister';

...
  start: (data: T, index: number) =>  unknown extends State ? HashMap : State;
...
```